packages/dds/merge-tree/src/test/obliterateOperations.ts

```diff
@@ -0,0 +1,201 @@
+
+import { annotateRange, type TestOperation } from "./mergeTreeOperationRunner.js";
+import type { TestClient } from "./testClient.js";
+
+export const posInField = (
+	client: TestClient,
+	pos: number,
+): { startPos: number; endPos: number } | undefined => {
+	if (
+		pos >= client.getLength() ||
+		(!Number.isInteger(Number(client.getText(pos, pos + 1))) &&
+			client.getText(pos, pos + 1) !== "{" &&
+			client.getText(pos, pos + 1) !== "}")
+	) {
+		return undefined;
+	}
+
+	let startPos = pos;
+	let endPos = pos;
+	// To find the start and end separators, walk backwards and forwards until the desired character is found.
+	while (
+		startPos > 0 &&
+		client.getText(startPos, startPos + 1) !== "{" &&
+		(client.getText(startPos, startPos + 1) === "}" ||
+			Number.isInteger(Number(client.getText(startPos, startPos + 1))))
+	) {
+		startPos--;
+	}
+
+	while (
+		endPos < client.getLength() &&
+		client.getText(endPos, endPos + 1) !== "}" &&
+		(client.getText(endPos, endPos + 1) === "{" ||
+			Number.isInteger(Number(client.getText(endPos, endPos + 1))))
+	) {
+		endPos++;
+	}
+
+	assert(startPos >= 0 && endPos <= client.getLength(), "Field endpoints are out of bounds");
+
+	return { startPos, endPos };
+};
+
+export const getFieldEndpoints = (
+	client: TestClient,
+	start: number,
+	end: number,
+): { startPos: number; endPos: number } | undefined => {
+	const startField = posInField(client, start);
+	const endField = posInField(client, end);
+
+	if (startField === undefined && endField === undefined) {
+		return undefined;
+	}
+	return startField ?? endField;
+};
+
+const generateFieldText = (client: TestClient, random: IRandom): string => {
+	const chunkLength = random.integer(1, 10);
+	return (client.longClientId!.codePointAt(0)! % 10).toString().repeat(chunkLength);
+};
+
+const insertFieldText: TestOperation = (
+	client: TestClient,
+	opStart: number,
+	opEnd: number,
+	random: IRandom,
+) => {
+	const text = generateFieldText(client, random);
+	return client.insertTextLocal(opStart, text);
+};
+
+export const insertField: TestOperation = (
+	client: TestClient,
+	opStart: number,
+	opEnd: number,
+	random: IRandom,
+) => {
+	const numberText = generateFieldText(client, random);
+	if (posInField(client, opStart) === undefined) {
+		return client.insertTextLocal(opStart, `{${numberText}}`);
+	}
+};
+
+export const obliterateField: TestOperation = (
+	client: TestClient,
+	opStart: number,
+	opEnd: number,
+	random: IRandom,
+) => {
+	const fieldEndpoints = getFieldEndpoints(client, opStart, opEnd);
+
+	let endISP: InteriorSequencePlace | undefined;
+	if (fieldEndpoints !== undefined) {
+		const { startPos, endPos } = fieldEndpoints;
+		// Obliterate text bewteen the separators, but avoid the case where the obliterate range is zero length.
+		if (endPos - startPos > 1) {
+			const op = client.obliterateRangeLocal(
+				{ pos: startPos + 1, side: Side.Before },
+				endISP ?? { pos: endPos - 1, side: Side.After },
```

Comment:
Am I misremembering how endpoint expansion works or should this be the following?
```suggestion
				{ pos: startPos, side: Side.After },
				endISP ?? { pos: endPos, side: Side.Before },
```

Reply:
That's how I would have expected it to work as well, but applying the suggestion led to a different set of failures (clients not having the same final state after applying all ops). This makes me think that these tests are hitting legitimate issues in the endpoint expansion code instead of issues with the test setup.

Reply:
I still think it makes more sense to use the endpoints without modification and update the side based on that information though, in terms of the test.

Reply:
I'm not sure what you mean. This test should be exercising endpoint expansion correctly. Any bugs hit are a separate issue. If it would be easier to discuss in a call, we can do that.